File: EpiAttendance.Api/Models/AttendanceRecord.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

namespace EpiAttendance.Api.Models;

public class AttendanceRecord
{
    [Key]
    public int Id { get; set; }
    [Required]
    public DateOnly Date { get; set; }
    [Required]
    public AttendanceType Type { get; set; }
    [Required]
    public string UserId { get; set; }
    public string? Notes { get; set; }
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: epi_attendance_app/lib/models/attendance_record.dart
import 'attendance_type.dart';

class AttendanceRecord {
  final int id;
  final DateTime date;
  final AttendanceType type;
  final String? notes;
  final DateTime createdAt;
  final DateTime? updatedAt;

  AttendanceRecord({
    required this.id,
    required this.date,
    required this.type,
    this.notes,
    required this.createdAt,
    this.updatedAt,
  });

  factory AttendanceRecord.fromJson(Map<String, dynamic> json) {
    return AttendanceRecord(
      id: json['id'] as int,
      date: DateTime.parse(json['date'] as String),
      type: AttendanceType.fromValue(json['type'] as int),
      notes: json['notes'] as String?,
      createdAt: DateTime.parse(json['createdAt'] as String),
      updatedAt: json['updatedAt'] != null
          ? DateTime.parse(json['updatedAt'] as String)
          : null,
    );
  }
}
